### PR TITLE
HPCC-26887 WsStore should not report misses as exceptions

### PR DESCRIPTION
**Pull request by rpastrana**
- Removes logic which forced an exception upon fetch miss
- Logs wsstore misses

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by rpastrana** [comment]
@GordonSmith as-is, a fetch miss renders an empty "Value" response:
in XML: <Value/>
in JSON: "Value": null

File: esp/services/ws_store/espstorelib/daliKVStore.cpp [hunk 1]
```diff
@@ -410,14 +410,19 @@ bool CDALIKVStore::fetch(const char * storename, const char * ns, const char * k
     {
         xpath.appendf("/%s", key);
         if(!storetree->hasProp(xpath.str()))
-            throw makeStringExceptionV(MSGAUD_user, -1, "DALI Keystore fetch: invalid key '%s' detected!", key);
-
-        value.set(storetree->queryProp(xpath.str()));
+        {
+            LOG(MCuserInfo,"DALI Keystore fetch: invalid key '%s' detected!", key);
```
Comment (wangkx) [inline esp/services/ws_store/espstorelib/daliKVStore.cpp:414]: Style: add a space after the first ','.

File: esp/services/ws_store/espstorelib/daliKVStore.cpp [hunk 2]
```diff
@@ -410,14 +410,19 @@ bool CDALIKVStore::fetch(const char * storename, const char * ns, const char * k
...
+        else
+        {
+            value.set(storetree->queryProp(xpath.str()));
+        }
 
         return value.str();
     }
     else
-        throw MakeStringException(-1, "DALI Keystore fetch: Namespace not provided!");
+        throw MakeStringException(-1, "DALI Keystore fetch: Key not provided!");
```
Comment (wangkx) [inline esp/services/ws_store/espstorelib/daliKVStore.cpp:425]: Minor: change the MakeStringException() to makeStringException() since you are changing this code line.

**Review by wangkx — COMMENTED** [review]
@rpastrana Changes look fine. 2 minor comments.

**Comment by rpastrana** [comment]
@wangkx please review latest change

**Comment by rpastrana** [comment]
@dcamper please let me know if this change breaks you. Also, does the ECL change make sense? Is there a better way to test for empty string in ECL?

**Review by wangkx — COMMENTED** [review]
@rpastrana the cpp code looks fine now. I have no knowledge about the ecl code.

**Comment by rpastrana** [comment]
Thanks @wangkx. Do you know if there's a way to prevent ESP from populating a response field?
For example, if wsstore::fetch(invalidkey), I'd like to ensure the response does not contain a value field.

**Comment by wangkx** [comment]
> Do you know if there's a way to prevent ESP from populating a response field? For example, if wsstore::fetch(invalidkey), I'd like to ensure the response does not contain a value field.

Please try nil_remove. For example, ESPresponse [nil_remove] UserQueryResponse.

**Comment by rpastrana** [comment]
@dcamper Dan, please review. I'll let you decide if the response should include a success flag for the sake of ECL-centric logic. Thanks.

**Review by dcamper — APPROVED** [review]
Approving only the ECL changes.

I think the change is minimal and unlikely to cause issues with existing users/installations.  If there is a problem, we'll raise a Jira and address it at that time.

File: esp/services/ws_store/ws_storeService.cpp [hunk 3]
```diff
@@ -312,8 +312,23 @@ bool CwsstoreEx::onFetch(IEspContext &context, IEspFetchRequest &req, IEspFetchR
...
+    {
+        m_storeProvider->fetch(storename, req.getNamespace(), req.getKey(), value, secuser.get(), !req.getUserSpecific());
+        resp.setValue(value.str());
+    }
+    catch(IException * e)
+    {
+        if (e->errorCode() == ECLWATCH_INVALID_QUERY_KEY)
+        {
+            StringBuffer msg;
+            LOG(MCuserInfo, "WsStore: %s", e->errorMessage(msg).str());
```
Comment (richardkchapman) [inline esp/services/ws_store/ws_storeService.cpp:325]: A bit concerned that this could lead to excessive logging (and potentially logging of sensitive information, depending on usage).

**Review by richardkchapman — CHANGES_REQUESTED** [review]

File: esp/services/ws_store/ws_storeService.cpp (hunk 3 above)
Comment (rpastrana) [reply esp/services/ws_store/ws_storeService.cpp:325]: That's definitely a valid concern which could apply to many logging statements. Perhaps it warrants adding a check-mark in the HPCC PR template to remind developers to be very mindful of the log output. and a full reassessment of logging levels and content.

As far as this particular change, suppressing the exception is intended to increase the log statement level from error to user-info. Depending on configuration it will appear less often now
I actually intended to output the offending key, and I will put that back in unless that in it self is one of your concerns.

**Review by richardkchapman — APPROVED** [review]

**Comment by richardkchapman** [comment]
@rpastrana Please squash
